casa/IO/test/tTypeIO.cc

```diff
@@ -131,34 +131,35 @@ void doIt (TypeIO* io)
 
 int main()
 {
-    RegularFileIO regularFileIO (Path("tTypeIO_tmp.dat"), ByteIO::New);
-    CanonicalIO canonicalIO (&regularFileIO);
+    std::shared_ptr<ByteIO> regularFileIO
+      (new RegularFileIO (Path("tTypeIO_tmp.dat"), ByteIO::New));
```

Comment:
```suggestion
    std::shared_ptr<RegularFileIO> regularFileIO
      (new RegularFileIO (Path("tTypeIO_tmp.dat"), ByteIO::New));
```
Additionally, you could use `std::make_shared` here.